src/mono/mono.proj

```diff
@@ -161,6 +163,50 @@
           IgnoreStandardErrorWarningFormat="true" />
   </Target>
 
+  <!-- Sets up WASI SDK if you don't have the WASI_SDK_PATH env variable set -->
+  <Target Name="ProvisionWasiSdk"
+          Condition="'$(ShouldProvisionWasiSdk)' == 'true' and '$(SkipMonoCrossJitConfigure)' != 'true'">
+    <ReadLinesFromFile File="$(MSBuildThisFileDirectory)/wasi/wasi-sdk-version.txt">
+      <Output TaskParameter="Lines" ItemName="_VersionLines" />
+    </ReadLinesFromFile>
+
+    <PropertyGroup>
+      <WASI_SDK_PATH>$(ProvisionWasiSdkDir)</WASI_SDK_PATH>
+      <WasiLocalPath>$([MSBuild]::NormalizeDirectory('$(MSBuildThisFileDirectory)', 'wasi'))</WasiLocalPath>
+      <WasiSdkVersion>%(_VersionLines.Identity)</WasiSdkVersion>
+      <WasiSdkUrl>https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-$(WasiSdkVersion)/wasi-sdk-$(WasiSdkVersion).0-linux.tar.gz</WasiSdkUrl>
+      <WasiSdkUrl Condition="'$(HostOS)' == 'osx'" >https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-$(WasiSdkVersion)/wasi-sdk-$(WasiSdkVersion).0-macos.tar.gz</WasiSdkUrl>
+      <WasiSdkUrl Condition="'$(HostOS)' == 'windows'" >https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-$(WasiSdkVersion)/wasi-sdk-$(WasiSdkVersion).0-mingw.tar.gz</WasiSdkUrl>
+    </PropertyGroup>
+
+    <RemoveDir Directories="$(WASI_SDK_PATH)" />
+    <Exec Command="rm -rf wasi-sdk-*/ wasi-sdk/ &amp;&amp; curl -L -o- $(WasiSdkUrl) | tar -xvz &amp;&amp; mv wasi-sdk-$(WasiSdkVersion).*/ wasi-sdk/ &amp;&amp; cp wasi-sdk-version.txt wasi-sdk/wasi-sdk-version.txt"
```

Comment:
I think these downloaded files should go to artefacts folder.

@akoeplinger thouhts?

Reply:
This duplicates how `emsdk`works, moving it to artifacts will mean you will have to download it every time when you `rm -rf artifacts` which I do after merge from main quite often. We provision wasmtime into artifacts folder, but that's much smaller package. 

There is also WASI_SDK_PATH variable, in which we could provide external location.

It would be nice to have it provisioned into user's home folder.

@radical thoughts ?

Reply:
I made the provisioning use the `WASI_SDK_PATH` and provision there even if it's not in the runtime folder.
I will switch the default from `src/mono/wasi/wasi-sdk` to `artifacts/wasi-sdk` in my next commits.

Reply:
I'll add changes for this from my branch to here.

Reply:
done